scipy/interpolate/_bsplines.py

```diff
@@ -2323,15 +2350,30 @@ def make_smoothing_spline(x, y, w=None, lam=None):
         raise ValueError('Regularization parameter should be non-negative')
 
     # solve the initial problem in the basis of natural splines
-    c = solve_banded((2, 2), X + lam * wE, y)
+    if np.ndim(lam) == 0:
+        c = solve_banded((2, 2), X + lam * wE, y)
+    elif np.ndim(lam) == 1:
+        # XXX: solve_banded does not suppport batched `ab` matrices; loop manually
+        c = np.empty((n, lam.shape[0]))
+        for i in range(lam.shape[0]):
+            c[:, i] = solve_banded((2, 2), X + lam[i] * wE, y[:, i])
+    else:
+        # this should not happen, ever
+        raise RuntimeError("Internal error, please report it to SciPy developers.")
+    c = c.reshape((c.shape[0], *y_shape1))
+
+    # hack: these are c[0], c[1] etc, shape-compatible with np.r_ below
+    c0, c1 = c[0:1, ...], c[1:2, ...]     # c[0], c[1]
+    cm0, cm1 = c[-1:-2:-1, ...], c[-2:-3:-1, ...]    # c[-1], c[-2]
+
     # move back to B-spline basis using equations (2.2.10) [4]
-    c_ = np.r_[c[0] * (t[5] + t[4] - 2 * t[3]) + c[1],
-               c[0] * (t[5] - t[3]) + c[1],
-               c[1: -1],
-               c[-1] * (t[-4] - t[-6]) + c[-2],
-               c[-1] * (2 * t[-4] - t[-5] - t[-6]) + c[-2]]
+    c_ = np.r_[c0 * (t[5] + t[4] - 2 * t[3]) + c1,
+               c0 * (t[5] - t[3]) + c1,
+               c[1: -1, ...],
+               cm0 * (t[-4] - t[-6]) + cm1,
+               cm0 * (2 * t[-4] - t[-5] - t[-6]) + cm1]
 
-    return BSpline.construct_fast(t, c_, 3)
+    return BSpline.construct_fast(t, c_, 3, axis=axis)
```

Comment:
Ah, so `make_smoothing_spline` just figures out the parameters with which to construct the `BSpline`.

If we've tested once that all the methods of `BSpline` have proper batch support, and we test that `make_smoothing_spline` produces a valid batched `BSpline`, we don't need to test _all_ the methods after producing a batch spline with `make_smoothing_spline`. So the tests are indeed overkill. I'll suggest a way of trimming them.

---

One thought here: we pass `axis` into `construct_fast` even though the data axis has been moved to `0`. I take it that `axis` is only used for transforming the output of the methods; `construct_fast` actually assumes that the data is already aligned along `axis` `0`?